lib/Mojo/WebSocketProxy/Config.pm

```diff
@@ -14,9 +14,9 @@ sub init {
     $self->{actions}  = {};
     $self->{backends} = {};
 
-    die 'Wrong parameter' if $in_config->{opened_connection} && ref($in_config->{opened_connection}) ne 'CODE';
-    die 'Wrong parameter' if $in_config->{finish_connection} && ref($in_config->{finish_connection}) ne 'CODE';
-    die 'Wrong parameter' if $in_config->{skip_check_sanity} && ref($in_config->{skip_check_sanity}) ne 'Regexp';
+    die 'Expected a coderef for opened_connection' if $in_config->{opened_connection} && ref($in_config->{opened_connection}) ne 'CODE';
+    die 'Expected a coderef for finish_connection' if $in_config->{finish_connection} && ref($in_config->{finish_connection}) ne 'CODE';
+    die 'Expected a regex for skip_check_sanity' if $in_config->{skip_check_sanity} && ref($in_config->{skip_check_sanity}) ne 'Regexp';
```

Comment:
General comment: I wonder if the general pattern of `ref $foo eq 'CODE'` ought to be changed in favour of `Ref::Util` - those tests are all efficient XS-or-custom-op code, so a bit more efficient than the native perl solution. Hardly an issue unique to this code, nor particularly important in this mere setup routine, but a general pattern we might want to consider for the codebase as a whole. Perhaps worthy of a new Trello card?

Reply:
Yes, I like that idea. We were already pulling in XS deps (via Guard.pm, since removed) so I don't see any backcompat issues with adding that in now. :+1: for the Trello card to pick up other cases in our code too.

Reply:
https://trello.com/c/zYLQTFRi/6610-consider-refutil